web/src/main/java/com/epam/esm/web/controller/CertificateController.java

```diff
@@ -0,0 +1,88 @@
+package com.epam.esm.web.controller;
+
+import com.epam.esm.service.CertificateService;
+import com.epam.esm.service.constant.CustomHttpStatus;
+import com.epam.esm.service.dto.CertificateDto;
+import com.epam.esm.service.dto.ErrorMessageDto;
+import com.epam.esm.service.exception.NotFoundException;
+import com.epam.esm.service.locale.LanguageManager;
+import com.epam.esm.service.locale.LocaleMessages;
+import com.epam.esm.web.hateoas.CertificateLinkBuilder;
+import com.epam.esm.web.hateoas.LinkBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/certificates")
+public class CertificateController {
+  private static final String DEFAULT_PAGE = "1";
+  private static final String DEFAULT_PAGE_SIZE = "10";
+
+  private final CertificateService certificateService;
+
+  @Autowired
+  public CertificateController(CertificateService certificateService) {
+    this.certificateService = certificateService;
+  }
+
+  @GetMapping
+  public HttpEntity<LinkBuilder<List<LinkBuilder<CertificateDto>>>> getAll(
+      @RequestParam(value = "sort", required = false) List<String> sortParams,
+      @RequestParam(value = "part", required = false) String partOfNameOrDescription,
```

Comment:
@Exqzore Why don’t you divide this parameter by two so we know exactly what to look for? Taking a map as an example. You can expand the search options.